### PR TITLE
Update sigstore dependencies

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Update sigstore dependencies

**Which issue(s) this PR fixes**:
Supersede #1 and #2.
Close #1 
Close #2

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependencies are updated:
- github.com/prometheus/client_golang: v1.14.0 -> v1.15.0
- github.com/sigstore/cosign: v1.13.1 -> v2.0.2
- github.com/sigstore/sigstore: v1.4.5 -> v1.6.3
- golang.org/x/tools: v.0.4.0 -> v0.8.0
- github.com/docker/docker v20.10.20+incompatible -> v23.0.3+incompatible
```

### CODE REVIEW

File: pkg/lakom/verifysignature/verifier.go [hunk 1]
```diff
@@ -145,8 +152,20 @@ func (r *cacheVerifier) Verify(ctx context.Context, image string, kcr utils.KeyC
 	return verified, nil
 }
 
-// IsNoMatchingSignature checks if error is of time github.com/sigstore/cosign/pkg/cosign.ErrNoMatchingSignatures.
+// IsNoMatchingSignature checks if error is of time github.com/sigstore/cosign/pkg/cosign.ErrNoMatchingSignaturesType.
```
[the remark this answers is not in the document]
Comment (dimityrmirchev) [reply pkg/lakom/verifysignature/verifier.go:155]: Suggestion: we can try to use Doc Links [ref](https://tip.golang.org/doc/comment). They were introduced with [go 1.19](https://tip.golang.org/doc/go1.19#go-doc)

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Some nits and a couple of questions. Other looks good

File: pkg/lakom/verifysignature/verifier.go [hunk 2]
```diff
@@ -77,12 +76,20 @@ func verify(ctx context.Context, imageRef name.Reference, keys []crypto.PublicKe
 			RegistryClientOpts: opts,
 			SigVerifier:        verifier,
 			ClaimVerifier:      cosign.SimpleClaimVerifier,
+			IgnoreSCT:          true,
+			IgnoreTlog:         true,
```
[the remark this answers is not in the document]
Comment (vpnachev) [reply pkg/lakom/verifysignature/verifier.go:80]: Transparent logs (TLog) are used to verify that there is a log entry about the image signature from when an image was signed. In gardener, we do not do not write logs in rekor, therefore this check is expected to fail. 

Regarding SCT, it is again related to rekor and my understanding is that it allows to check the tlog without querying the tlog system. More info can be found at https://docs.sigstore.dev/cosign/verify/#verify-a-signature-was-added-to-the-transparency-log. 

Anyway, bot features were disabled with the v1 version, with v2 - they are by default enabled and we need to explicitly disable them.

File: pkg/lakom/verifysignature/verifier.go [hunk 3]
```diff
@@ -77,12 +76,20 @@ func verify(ctx context.Context, imageRef name.Reference, keys []crypto.PublicKe
...
 		})
 		if err != nil {
+			if IsNoSignaturesFound(err) {
+				log.Info("no signatures found for the image", "error", err.Error())
```
[the remark this answers is not in the document]
Comment (vpnachev) [reply pkg/lakom/verifysignature/verifier.go:84]: The image is injected into the logger here https://github.com/gardener/gardener-extension-shoot-lakom-service/blob/ad6852939c9e4cecb42c366f4e1c7ada2a19fbbc/pkg/lakom/verifysignature/admission.go#L224-L225

Here is an example  (`make dev-setup; make start-lakom; k run alpine --image=alpine:3.13`):
```json
{
    "level": "info",
    "ts": "2023-05-10T18:06:05.777+0300",
    "logger": "cosign-signature-verifier",
    "msg": "no signatures found for the image",
    "pod": "default/alpine",
    "containers": "alpine",
    "image": "index.docker.io/library/alpine@sha256:469b6e04ee185740477efa44ed5bdd64a07bbdd6c7e5f5d169e540889597b911",
    "containerName": "alpine",
    "cosignKeyID": 0,
    "error": "no signatures found for image"
}
```

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**